app/components/github-connect.js

```diff
@@ -0,0 +1,35 @@
+import Ember from 'ember';
+
+const {
+  Component,
+  get,
+  set
+} = Ember;
+const baseUrl = 'https://github.com/login/oauth/authorize';
+
+/**
+  The github-connect component is used to to connect to github
+
+  ## default usage
+
+  ```handlebars
+    {{github-connect clientId=clientId scope=scope state=state redirectUri=redirectUri}}
+  ```
+
+  @module Component
+  @class github-connect
+  @extends Ember.Component
+  @public
+ */
+
+export default Component.extend({
+  tagName: 'a',
+  classNames: ['button', 'default'],
+  attributeBindings: ['url:href'],
+
+  init() {
+    this._super(...arguments);
+    this.getProperties('clientId', 'scope', 'state', 'redirectUri');
+    set(this, 'url', `${baseUrl}?scope=${get(this, 'scope')}&client_id=${get(this, 'clientId')}&state=${get(this, 'state')}&redirect_uri=${get(this, 'redirectUri')}`);
```

Comment:
This would be written as 

```javascript
let { clientId, scope, state, redirectUri } = getProperties(this, 'clientId', 'scope', 'state', 'redirectUri');
set(this, 'url', `${baseUrl}?scope=${scope)}&client_id=${clientId}&state=${state}&redirect_uri=${redirectUri}`);
```

Also, would you mind declaring these properties as default nulls in the component:

```Javascript
...
attributeBindings: ['url:href'],

clientId: null,
redirectUri: null,
scope: null,
state: null,
...
```

We like to do that, so that it's at a glance clear which properties the component relies on.